packages/composite-checkout/src/components/checkout-steps.js

```diff
@@ -799,7 +821,7 @@ function getStepNumberFromUrl() {
 		const stepNumber = parts.length > 1 ? parts[ 1 ] : 1;
 		return parseInt( stepNumber, 10 );
 	}
-	return 1;
+	return null;
```

Comment:
Without this returning `1`, is there a way to get to the first step without having to add `#step1` on the hash?

The scenario I'm thinking of is:

- User visits checkout without autocomplete of contact details.
- User fills out contact details and presses Continue to go to step 2.
- User pressed Back in the browser; the URL changes to remove `#step2`.
- `useChangeStepNumberForUrl` triggers a hashchange and calls this function to figure out the new step.

With this change, I _think_ it means that the current step will remain at step 2.

Reply:
I was able to reproduce the scenario above, so I returned the hash navigation to its old state and used `scrollIntoView` instead.